Deduplicate example code

zenflow opened the pull request:
Closes #27

@limonte So I deduplicated 3 code examples to start. Two sync and one async. Let me know if you have any issues before I deduplicate the rest, following this pattern.

The examples look & behave the exact same as before, with the exception that the async wrapper in codepen is changed to use an anonymous arrow function. In other words, `async function getIpAddress () {`  -> `async () => {`. This wrapper is suitable for all of the async examples.

limonte commented:
Good day @zenflow 

Do you need help on this one? Are you stuck on something or it's a lack of time?

zenflow commented:
@limonte sorry for the delay.. I'm not stuck on anything, it's just a lack of time.. work is demanding and I'm travelling & exploring cities at the same time.. 

Feel free to finish off the PR if you're tired of waiting. It should be fairly straight-forward from here, and if you run into any problems I am here to help solve them.

Otherwise hopefully I will be able to finish it off this coming weekend (and will be happy to do so).

limonte commented:
Thank you for the prompt reply @zenflow 

This PR can wait for your free time as much as needed, please continue when you have time and feel like it.

zenflow commented:
I rebased the work onto the current `master` and removed the changes to `dist/bundle.js`

zenflow commented:
Rebased again. Will wait for #86 to be cleared up (i.e. fix the build) and then I will finish this off (maybe within 3 months of me starting it 😄)

limonte commented:
Thank you @zenflow!

#86 is resolved now, sorry about that inconvenience!

zenflow commented:
Ok, I think this is ready now. I rebased again, finished deduplicating the remaining example code, fixed any eslint errors in `example/functions.js` (e87faeb), and tweaked a bit of the original code I did in this PR (7688bfe).

I used the *displayed* example code (as opposed to the *executed* example code) and kept an eye out for discrepancies. I didn't find many, but these are the discrepancies I found and what I did:
- rtl example - I dropped the `target: document.getElementById('rtl-container')` option missing from the displayed function. I'm not really sure what the point of it is in this example. Feel free to add it back with a comment explaining why it's there and that it's not required (as per https://github.com/sweetalert2/sweetalert2.github.io/issues/27#issuecomment-477095043) 
- textareaInput example - I kept the `inputAttributes: { 'aria-label': 'Type your message here' }` option missing from the displayed function (found in executed function). (To get it to not make the block much wider, I just had to divide it over 3 lines)
- multipleInputs example - I dropped the `placeholder` attributes on input elements, missing from the displayed function (found in executed function). (Feel free to add it back and refactor it to get it to not make the block much wider)
- timer example - I used the `html` option from the executed function (different from that of the displayed function) using the correct word ("millseconds") as per #1558

limonte commented:
Amazing work! Thank you so much @zenflow!

`target` is needed in the RTL example in order to show the close button properly (it'll be on the right side if a popup doesn't have a container with `dir="rtl"` or `direction: rtl`:

![image](https://user-images.githubusercontent.com/6059356/62617872-8baae880-b91b-11e9-81d7-9adf110e812d.png)

limonte commented on index.html:
```diff
@@ -149,434 +148,175 @@ <h3 id="examples">More examples</h3>
+      <pre data-example-id="timer"></pre>
     </li>
 
-    <li class="rtl" id="rtl">
```
ids are used as anchors, anchors are handy when it's needed to share a link to a particular example (e.g. https://github.com/sweetalert2/sweetalert2.github.io/blob/master/index.html#L1128)

What's the reson for dropping them in this PR?

limonte replied:
@zenflow FYI I reverted ids, other than that all is well :rocket:

zenflow commented:
Thanks for getting this merged promptly @limonte ! Yay!

Sorry you had to add the `id` attributes back! I removed them because I thought they were only being used for identifying elements for adding event handlers, or unused. There were no hashtag links in the document to any of these elements. 

The `// do-not-show` is a great idea for an escape-hatch for when we want something hidden from the displayed code, but I still think we should try to avoid doing that since it's kinda deceptive. Users are expecting the displayed code to produce the same effects that they see when they click the button to execute the example. IMO it could be avoided in this case by adding a comment (in the code or in the html description) to explain why that line is there and say that it's optional. But I'll leave that with you. I'm pretty happy with the state of things now!

limonte commented:
Agree with you @zenflow that we should use `// do-not-show` as less as possible, but in the RTL example I believe that line even with the comment will confuse readers more than help.

Thank you again for the great contribution!
